schemas/iso19139/src/main/plugin/iso19139/index-fields/index.xsl

```diff
@@ -768,22 +767,37 @@
         <xsl:for-each select="gmd:transferOptions/*/
                                 gmd:onLine/*[gmd:linkage/gmd:URL != '']">
 
-          <xsl:variable name="protocol" select="gmd:protocol/gco:CharacterString/text()"/>
+          <xsl:variable name="protocol"
+                        select="gmd:protocol/gco:CharacterString/text()"/>
+          <xsl:variable name="linkName"
+                        select="replace(gmd:name/gco:CharacterString/text(),
+                                              $doubleQuote, $escapedDoubleQuote)"/>
 
           <linkUrl>
             <xsl:value-of select="gmd:linkage/gmd:URL"/>
           </linkUrl>
           <linkProtocol>
             <xsl:value-of select="$protocol"/>
           </linkProtocol>
-          <link>
-            <xsl:value-of select="gmd:protocol/*/text()"/>
-            <xsl:text>|</xsl:text>
+          <xsl:element name="linkUrlProtocol{replace($protocol, '[^a-zA-Z0-9]', '')}">
```

Comment:
Is that really necessary? Looking at a document in ES, I see links indexed in 4 ways:
* `link` array (full objects)
* `linkUrl` array
* `linkProtocol` array
* `linkUrlProtocol<INSERT_PROTOCOL_HERE>` array

Not an ES expert but I would have thought that indexing only the first way would be enough (simpler, more readable, less maintenance...)

Reply:
eg. Kibana does not support aggregation on objects properties so the link field can't be used.
So the link object sounds the one to keep on the long run, will be used by the main UI. Other fields (similar to the one we have in Lucene) can be used in dashboards. Object types allows to do nested query also but this can't be used in Kibana.

So both type of fields are needed for now.

On the main ES branch, maybe we should have a doc of all fields in the index with docs (should be in https://github.com/geonetwork/core-geonetwork/blob/es/web/src/main/webapp/WEB-INF/data/config/index/records.json)